File: vuebugtrackerproject.client/src/classes/DTOs/BugFilterDTO.ts
import Severity from "@/enumConsts/Severity";
import Status from "@/enumConsts/Status";

// DTO for filtering a project's bug.
export default class BugFilterDTO{
  // Search query for the summary of the bug.
  summary: string = "";
  // Decides whether to filter bugs on date created or
  // last updated.
  dateSearch: number = 0;
  // The starting date search range.
	dateFrom: string | null = null;
	// The end date search range.
	dateEnd: string | null = null;
  // The username of whoever created the bug.
  creatorName: string = "";
  // Filters bugs based on severity. These are a set of
  // integers representing "low", "medium" and "high".
  severityValues: number[] = [Severity.Low, Severity.Medium, Severity.High];
  // Filters bugs based on whether they are open or closed.
  // These are a set of integers representing "open" and "closed".
  statusValues: number[] = [Status.Closed, Status.Open];
	// Determines how the items will be sorted.
	sortType: number = 0;
	// Determines whether the items are sorted in ascending or
	// descending order.
	sortOrder: number = 0;
  // The page of bugs that will be returned.
  pageNumber: number = 1;
  // The ID of the project the bugs belong to.
  projectID: string = "";
}

File: VueBugTrackerProject.Classes/DTOs/BugFilterDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace VueBugTrackerProject.Classes
{
    /// <summary>
    /// DTO for filtering a project's bug.
    /// </summary>
    public class BugFilterDTO
    {
        /// <summary>
        /// Search query for the summary of the bug.
        /// </summary>
        public string Summary { get; set; }

        /// <summary>
        /// Decides whether to filter bugs on date created or last updated.
        /// </summary>
        public DateSearch DateSearch { get; set; }

        /// <summary>
        /// The starting date search range.
        /// </summary>
        public DateTime? DateFrom { get; set; }

        /// <summary>
        /// The end date search range.
        /// </summary>
        public DateTime? DateEnd { get; set; }

        /// <summary>
        /// The username of whoever created the bug.
        /// </summary>
        public string CreatorName { get; set; }

        /// <summary>
        /// Filters bugs based on severity. These are a set of 3 true/false values
        /// in order of "low", "medium" and "high".
        /// </summary>
        public List<Severity> SeverityValues { get; set; }

        /// <summary>
        /// Filters bugs based on whether they are open or closed. These are a set of 2
        /// true/false values in order of "open" and "closed".
        /// </summary>
        public List<Status> StatusValues { get; set; }

        /// <summary>
        /// Determines how the items will be sorted.
        /// </summary>
        public SortType SortType { get; set; }

        /// <summary>
        /// Determines whether the items are sorted in ascending or descending order.
        /// </summary>
        public SortOrder SortOrder { get; set; }

        /// <summary>
        /// The page of bugs that will be returned.
        /// </summary>
        public int PageNumber { get; set; }

        /// <summary>
        /// The ID of the project the bugs belong to.
        /// </summary>
        public string ProjectID { get; set; }
    }
}
